### PR TITLE
BAVL-128 updates to support sending emails to prisons on creation of booking.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
These changes add in additional emails on the creation journey for prisons.

Two new prison emails have been added, one where we have a contact for a court and one where we don't have a court contact.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/config/EmailConfiguration.kt [hunk 1]
```diff
@@ -54,13 +58,13 @@ abstract class Email(
...
   date: LocalDate = LocalDate.now(),
-  comments: String? = "",
+  comments: String? = null,
 ) {
   private val common = mapOf(
     "date" to date.toMediumFormatStyle(),
     "prisonerName" to prisonerFirstName.plus(" $prisonerLastName"),
     "offenderNo" to prisonerNumber,
-    "comments" to comments,
+    "comments" to (comments ?: "None entered"),
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/config/EmailConfiguration.kt:67]: This address the null comments in email issue spotted during integration with the UI and API.

File: src/test/resources/application-test.yml [hunk 2]
```diff
@@ -1,5 +1,8 @@
 spring:
 
+  main:
+    allow-bean-definition-overriding: true
```
Comment (matt-crowson-moj) [inline src/test/resources/application-test.yml:4]: This is required for the the test configuration class TestEmailConfiguration.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/BookingFacade.kt [hunk 3]
```diff
@@ -47,42 +47,80 @@ class BookingFacade(
...
+            templateName = templateId,
+            reason = "New court booking request",
+          ),
+        )
+      }.onFailure { log.info("BOOKINGS: Failed to send new court booking email.") }
+    }
   }
 
+  private fun Collection<BookingContact>.primaryCourtContact() =
+    allContactsWithAnEmailAddress().singleOrNull { it.contactType == ContactType.COURT && it.primaryContact }
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/BookingFacade.kt:122]: I did wonder about expanding the BookingContactsService to provide such things but thought until we need it elsewhere then perhaps no real need as of yet.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/GovNotifyEmailService.kt [hunk 4]
```diff
@@ -55,3 +57,49 @@ class CourtNewBookingEmail(
...
+  mainAppointmentInfo: String,
+  postAppointmentInfo: String?,
+  comments: String?,
+) : Email(address, prisonerFirstName, prisonerLastName, prisonerNumber, date, comments) {
+  init {
+    addPersonalisation("court", court)
+    addPersonalisation("prison", prison)
+    addPersonalisation("preAppointmentInfo", preAppointmentInfo ?: "Not required")
+    addPersonalisation("mainAppointmentInfo", mainAppointmentInfo)
+    addPersonalisation("postAppointmentInfo", postAppointmentInfo ?: "Not required")
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/GovNotifyEmailService.kt:103]: I did wonder whether we need to replicate all these 22 templates, or whether we could make them "smarter" - with replaceable sections based on which data we had? Depends on the differences between the templates though.

Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/GovNotifyEmailService.kt:103]: Maybe copy for now, and exploit any commonalities when we see them?

File: src/test/resources/application-test.yml (hunk 2 above)
Comment (timharrison-moj) [reply src/test/resources/application-test.yml:4]: But what does it all mean, Matt? Are we overriding tests with a different bean definition?

**Review by timharrison-moj — APPROVED** [review]
Very good. LGTM

File: src/test/resources/application-test.yml (hunk 2 above)
Comment (matt-crowson-moj) [reply src/test/resources/application-test.yml:4]: No just in the one integration test for email testing.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/GovNotifyEmailService.kt (hunk 4 above)
Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/GovNotifyEmailService.kt:103]: Yeah 22 is a lot of templates.  Keen to refine/slim down.

**Pull request merged**